pkg/ds/daemon_set_test.go

```diff
@@ -418,12 +444,37 @@ func TestPublishToReplication(t *testing.T) {
 	quitCh := make(chan struct{})
 	updatedCh := make(chan *ds_fields.DaemonSet)
 	deletedCh := make(chan *ds_fields.DaemonSet)
-	defer close(quitCh)
-	defer close(updatedCh)
-	defer close(deletedCh)
 	desiresErrCh := ds.WatchDesires(quitCh, updatedCh, deletedCh)
 	dsChangesErrCh := watchDSChanges(ds, dsStore, quitCh, updatedCh, deletedCh)
 
+	// We need to be careful when shutting down in order to avoid data
+	// races, since updatedCh and deletedCh are shared between two
+	// functions (WatchDesires() and watchDSChanges()). We need to make
+	// sure that we don't close updatedCh or deletedCh until their
+	// respective output channels have closed which indicates that they
+	// won't try to send any more values
+	defer func() {
```

Comment:
bonus: i think this is the cause of the `panic: send on closed channel` flake we have